### ISSUE TITLE
High DPI / retina screens

### ISSUE THREAD

**Issue by s9w** [s9w/font_compare#3]
There were mentions of rendering issues with Retina / high DPI screens in the HN thread: https://news.ycombinator.com/item?id=9998193

I don't have a 4k Monitor or something equivalent to test. My attempt at fixing this with the srcset attribute is hosted here: http://www.eatginseng.com/font_testing/ . It delivers an upscaled image to the "2x" target. Note that the zoom function in the compare part has no effect for those screens since that would require even more images (a 4x zoom). Maybe I'll add that later.

If someone with retina and/or 4k displays could report if the images are still too small and/or blurry I would be very glad.

**Comment by be5invis** [comment]
+1 for that.

**Comment by pbnjay** [comment]
looks the same. you can't upscale 1x to 2x, you have to render 2x pixels.

Since you're doing screencaps in windows, i'm guessing the best way you can attack it is to render at double the font point size. Hope that helps!

**Comment by s9w** [comment]
So still blurry? Could you make a screenshot? As long as the browser is obeying the srcset stuff it should use the point-upscaled version.

I agree that the best way would be to actually provide much bigger font sizes as you would use on such a high resolution. But font sizes are fixed for bitmap fonts, so I'll have to think about that. Maybe 2x-upscale bitmap fonts and properly screenshot the others.

**Comment by pbnjay** [comment]
current site:
<img width="318" alt="screen shot 2015-08-07 at 3 32 16 pm" src="https://cloud.githubusercontent.com/assets/161552/9144218/88509694-3d19-11e5-9ae7-fc9adee5aeae.png">

test site
<img width="301" alt="screen shot 2015-08-07 at 3 31 58 pm" src="https://cloud.githubusercontent.com/assets/161552/9144219/885a62b4-3d19-11e5-91f4-62991c0a9497.png">

**Comment by pbnjay** [comment]
yeah you won't be able to do anything with bitmap fonts, but i dont think anyone with high DPI will care about them anyway

**Comment by s9w** [comment]
By now both sites have the same upscaling. Thank you! That's actually working exactly as intended, even if it's only a makeshift solution.

Yeah I agree on the bitmap fonts. I'll probably add another set of much bigger font sizes as an option then. Any size preferences?

**Comment by be5invis** [comment]
Maybe you can add "real retina" example images besides pixelated ones.

**Comment by s9w** [comment]
I've now added a big font size for every non-bitmap font. There's a global switch in overview mode and is available as a normal size during comparing. It's 20-22 for most fonts, the exact values were hand-picked so all fonts are roughly the same size. 

All DPIs are now forced to the same image, so high dpi people should just pick the big size and it should be fine. Does that work for everyone?

**Comment by oullah** [comment]
Size is OK but the text looks blurry on my machine. 
Yoga 2 Pro - 3200 x 1800
Windows 10
Firefox

**Comment by s9w** [comment]
What browser? Could you post a screenshot?

edit: oh nevermind about the browser

**Comment by jcelerier** [comment]
Here's how it looks for me on a macbook pro retina with archlinux under firefox : http://imgur.com/stOwbqy,krylCmk

**Comment by s9w** [comment]
Well, that's obviously not how it's supposed to look. And at this point I'm kind of out of ideas. On high DPI screens, browsers for some reason like to double the size of images. I've implemented a forced use of non-scaled images with the srcset attribute, but some browsers seem to ignore this.

I've even tried using a canvas to display images, but it's slow and sometimes not at all reacting to a change of image links.

But thanks for the reports, I'll have to do some more digging to come up with a final solution for this.
